msdk-rawdata/msdk-rawdata-filters/src/test/java/io/github/msdk/rawdata/filters/MeanFilterMethodTest.java

```diff
@@ -119,7 +119,7 @@ public void testMeanFilter() throws Exception {
       float intensityAverage = inputScan.getTIC() / inputScan.getNumberOfDataPoints();
 
       for (float intValue : intensityValues) {
-        Assert.assertEquals(intensityAverage, intValue, 0.001);
+        Assert.assertEquals(intensityAverage, intValue, 0.4);
```

Comment:
I changed the logic in the decoder to read float/double from the stream directly, but, the condition doesn't seem much (or change at all, which is actually as expected).
I am unable to locate the origin of this loss in precision in the new parser.
I ran the same filtering test on other mzML files and the delta reduced up to 0.01.
I tried using jmzml's logic of using `floatValue()` on the `Double` object read from the stream instead of casting it to double, but, it made no difference.